### PR TITLE
Fix issue DPTP-3839 Migrate to OpenShift Logging

### PR DESCRIPTION
**Pull request by bear-redhat**

### CODE REVIEW

**Comment by bear-redhat** [comment]
The file needs to apply twice to be success.

**Review by hongkailiu — APPROVED** [review]

**Comment by hongkailiu** [comment]
One question that is still not addressed from our slack discussion.
- Do we need to deploy splunk on OSD clusters (since SRE did it already)?

Since we start to have common_ocp and common managed, how do we make the symlinks when a new cluster joins? Manual work? I think those should be done via cluster-init.

/lgtm cancel

File: clusters/build-clusters/common_managed/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+This directory is used to store common objects we used only in managed clusters (e.g., build01-10).
```
Comment (hongkailiu) [inline clusters/build-clusters/common_managed/README.md:1]: Could you also give an example of the negative case?
Would a hypershift cluster counted as managed? I know there isnt one today. Could be one tomorrow.

Comment (hongkailiu) [reply clusters/build-clusters/common_managed/README.md:1]: Are the managed clusters defined here too?
https://github.com/openshift/release/blob/master/clusters/build-clusters/_cluster-init.yaml

Comment (bear-redhat) [reply clusters/build-clusters/common_managed/README.md:1]: Hypershift clusters are just type of clusters similar as OSD vs OCP.
Yes.

**Comment by bear-redhat** [comment]
> One question that is still not addressed from our slack discussion.
> 
> * Do we need to deploy splunk on OSD clusters (since SRE did it already)?
> 
> Since we start to have common_ocp and common managed, how do we make the symlinks when a new cluster joins? Manual work? I think those should be done via cluster-init.
> 
> /lgtm cancel

Manual for now.

File: clusters/build-clusters/common_ocp/openshift-logging-operator.yaml
Comment (hongkailiu) [inline clusters/build-clusters/common_ocp/openshift-logging-operator.yaml:1]: Could we also add a readme to this folder?
Explain
- how to determine if a cluster is an OCP cluster
- any manual work needs to do when a new cluster joins.

File: clusters/build-clusters/common_managed/README.md (hunk 1 above)
Comment (hongkailiu) [reply clusters/build-clusters/common_managed/README.md:1]: Could a hypershift cluster be considered managed?

Comment (hongkailiu) [reply clusters/build-clusters/common_managed/README.md:1]: This one is still unaddressed.
> Could you also give an example of the negative case?

File: clusters/build-clusters/common_ocp/openshift-logging-operator.yaml
Comment (bear-redhat) [reply clusters/build-clusters/common_ocp/openshift-logging-operator.yaml:1]: I added the README.md to `common_ocp` as well

**Comment by bear-redhat** [comment]
> > Manual for now.
> 
> Could we clarify that in the readme?

it literally says `This directory should be symlinked to all OCP clusters directories.`

**Comment by hongkailiu** [comment]
> it literally says `This directory should be symlinked to all OCP clusters directories.`

But it is not clear to me if it a manual work for now.

File: clusters/build-clusters/common_managed/README.md (hunk 1 above)
Comment (bear-redhat) [reply clusters/build-clusters/common_managed/README.md:1]: why we give negative case there? I think is just make things more confusing.

Comment (bear-redhat) [reply clusters/build-clusters/common_managed/README.md:1]: > Could a hypershift cluster be considered managed?
Same as asking `could a OSD cluster be considered managed`? Its just depends on how you configure the cluster.

**Review by jmguzik — APPROVED** [review]

File: clusters/build-clusters/common_managed/README.md (hunk 1 above)
Comment (hongkailiu) [reply clusters/build-clusters/common_managed/README.md:1]: Let me rephrase:
I would like to know if this folder should be applied to a given cluster.
At the moment, we use this algorithm to determine: OCP, OKD, OSD.

```
oc get svc -n openshift-monitoring configure-alertmanager-operator
```
If the above `svc` exists, then OSD.
Otherwise, 

```console
version=$(oc get clusterversion version --context build01 -o yaml| yq '.status.history[0].version')
```
If "$version" contains 'okd', then the cluster is OKD, otherwise, OSD.

So ... now we have a new type "managed" and another one "hosted". How do we extend the above algorithm to determine if or not a cluster belongs to the type?

**Comment by hongkailiu** [comment]
I replied and noticed that it has been approved by @jmguzik .

The goals I am trying to achieve are
- What to do when a new cluster joins. Not only to you and me but also to the others in the team.
- The PR fixes a broken component (splunk) which is good but it also creates some manual steps to new clusters. I want to make it clear in the PR.
